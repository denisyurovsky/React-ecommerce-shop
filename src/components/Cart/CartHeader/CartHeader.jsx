import Button from '@mui/material/Button';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';

import { selectCart } from '../../../store/cart/cartSlice';
import { Title } from '../../Title/Title';

import styles from './CartHeader.module.scss';

export const CartHeader = ({ openModal }) => {
  const cart = useSelector(selectCart);

  return (
    <div className={styles.header}>
      <Title>Cart</Title>
      {(cart.sellers && Object.keys(cart.sellers).length) !== 0 && (
        <Button
          variant="outlined"
          className={styles.emptyButton}
          onClick={openModal}
        >
          Empty cart
        </Button>
      )}
    </div>
  );
};

CartHeader.propTypes = {
  openModal: PropTypes.func,
};
