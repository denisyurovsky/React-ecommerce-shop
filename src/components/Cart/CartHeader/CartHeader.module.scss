@import "../../../assets/styles/colors.scss";
@import "../../../assets/styles/breakpoints.scss";

.header {
  display: flex;
  height: 100px;
  justify-content: space-between;
  align-items: center;
}

@media all and (max-width: $sm) {
  .header {
    padding: 0 !important;
  }
}

.emptyButton {
  color: $greyDark !important;
  border-color: $greyDark !important;
  text-transform: capitalize;
}

.emptyButton:hover {
  border-color: $greyDark;
}
