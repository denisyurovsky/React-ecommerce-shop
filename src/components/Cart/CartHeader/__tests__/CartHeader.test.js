import React from 'react';

import render from '../../../../test-utils/renderWithStore';
import { CartHeader } from '../CartHeader';

describe('CartHeader component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(<CartHeader openModal={() => {}} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
