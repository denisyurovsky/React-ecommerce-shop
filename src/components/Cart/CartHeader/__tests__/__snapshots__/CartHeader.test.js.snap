// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h2
      class="header"
      data-testid="title"
    >
      Cart
    </h2>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root emptyButton css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Empty cart
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
